apps/server/src/modules/system/repo/system.repo.ts

```diff
@@ -23,6 +23,20 @@ export class SystemRepo {
 		return domainObject;
 	}
 
+	public async findAll(): Promise<System[]> {
+		const entities: SystemEntity[] = await this.em.find(SystemEntity, {});
+
+		const domainObjects: System[] = entities.map((entity) => {
+			const props: SystemProps = SystemDomainMapper.mapEntityToDomainObjectProperties(entity);
+
+			const domainObject: System = new System(props);
+
+			return domainObject;
+		});
+
+		return domainObjects;
+	}
+
```

Comment:
Auch hier kann es viel zu viel zurück liefern.
Ich persönlich wurde es anderes umbauen. 
```
Controller.getLDAPSchools-> SchoolUC ->SchoolService -> SystemService -> getLDAPSystems() ->
SystemRepo.findByTyp(type)   ->getScholsBySystemIds(systemId) -> SchoolRepo.findBySystemIds

```

Reply:
Ich hab es so gebaut, damit die Methode `isEligibleForLdapLogin` im DO stehen kann, weil das ja sozusagen eine "Business Rule" ist. Aber wenn es viele Systems gibt, wäre es wahrscheinlich besser, sie gleich in der Datenbank zu filtern. Hast du mal nachgeguckt, wie viele Systems es auf Prod gibt?

Reply:
Habs so umgebaut, wie du vorschlägst.